In this project we are going to analyze the dateset of heart disease patients

In [61]:
import numpy as np
import pandas as pd

In [62]:
data = pd.read_csv('heart_disease_dataset.csv')
data

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
0,75,Female,228,119,66,Current,Heavy,1,No,No,Yes,8,119,Yes,Atypical Angina,1
1,48,Male,204,165,62,Current,NaN,5,No,No,No,9,70,Yes,Typical Angina,0
2,53,Male,234,91,67,Never,Heavy,3,Yes,No,Yes,5,196,Yes,Atypical Angina,1
3,69,Female,192,90,72,Current,NaN,4,No,Yes,No,7,107,Yes,Non-anginal Pain,0
4,62,Female,172,163,93,Never,NaN,6,No,Yes,No,2,183,Yes,Asymptomatic,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,56,Female,269,111,86,Never,Heavy,5,No,Yes,Yes,10,120,No,Non-anginal Pain,1
996,78,Female,334,145,76,Never,NaN,6,No,No,No,10,196,Yes,Typical Angina,1
997,79,Male,151,179,81,Never,Moderate,4,Yes,No,Yes,8,189,Yes,Asymptomatic,0
998,60,Female,326,151,68,Former,NaN,8,Yes,Yes,No,5,174,Yes,Atypical Angina,1


In [63]:
data.head(10)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
0,75,Female,228,119,66,Current,Heavy,1,No,No,Yes,8,119,Yes,Atypical Angina,1
1,48,Male,204,165,62,Current,NaN,5,No,No,No,9,70,Yes,Typical Angina,0
2,53,Male,234,91,67,Never,Heavy,3,Yes,No,Yes,5,196,Yes,Atypical Angina,1
3,69,Female,192,90,72,Current,NaN,4,No,Yes,No,7,107,Yes,Non-anginal Pain,0
4,62,Female,172,163,93,Never,NaN,6,No,Yes,No,2,183,Yes,Asymptomatic,0
5,77,Male,309,110,73,Never,NaN,0,No,Yes,Yes,4,122,Yes,Asymptomatic,1
6,64,Female,211,105,86,Former,Heavy,8,Yes,Yes,Yes,2,120,No,Typical Angina,1
7,60,Female,208,148,83,Never,Moderate,4,No,Yes,Yes,2,113,Yes,Asymptomatic,1
8,37,Female,317,137,66,Current,Heavy,3,No,Yes,Yes,5,114,No,Non-anginal Pain,0
9,63,Male,204,141,68,Former,Heavy,8,No,Yes,No,3,107,No,Asymptomatic,1


In [64]:
data.tail(10)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
990,54,Male,304,112,86,Former,NaN,8,Yes,No,Yes,6,111,No,Asymptomatic,1
991,26,Female,215,100,74,Never,Heavy,7,No,Yes,No,10,135,No,Atypical Angina,0
992,28,Female,220,102,73,Current,Moderate,7,Yes,Yes,Yes,10,102,No,Typical Angina,0
993,70,Female,346,110,62,Never,NaN,1,Yes,Yes,No,8,130,Yes,Asymptomatic,1
994,52,Male,248,159,76,Former,Moderate,9,No,Yes,Yes,2,152,Yes,Asymptomatic,1
995,56,Female,269,111,86,Never,Heavy,5,No,Yes,Yes,10,120,No,Non-anginal Pain,1
996,78,Female,334,145,76,Never,NaN,6,No,No,No,10,196,Yes,Typical Angina,1
997,79,Male,151,179,81,Never,Moderate,4,Yes,No,Yes,8,189,Yes,Asymptomatic,0
998,60,Female,326,151,68,Former,NaN,8,Yes,Yes,No,5,174,Yes,Atypical Angina,1
999,53,Male,226,116,82,Current,NaN,6,No,No,Yes,5,161,Yes,Asymptomatic,1


In [65]:
data.sample(5)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
365,40,Female,332,144,96,Never,NaN,1,No,Yes,No,2,99,No,Non-anginal Pain,0
281,35,Male,171,172,70,Former,Heavy,7,Yes,Yes,No,2,107,Yes,Typical Angina,0
787,62,Female,200,170,66,Current,NaN,5,No,No,No,1,86,Yes,Asymptomatic,0
250,51,Female,266,102,86,Never,NaN,2,Yes,Yes,Yes,2,167,Yes,Atypical Angina,1
524,48,Female,187,132,72,Former,Heavy,1,Yes,Yes,Yes,3,198,No,Non-anginal Pain,0


🔵 Section 1 — Data Understanding (Q1–Q8)

1. What is the shape of the dataset (rows and columns)?

In [66]:
data.shape

(1000, 16)

Dataset have 1000 rows and 16 columns

2. What are the column names and their data types?

In [67]:
data.columns

Index(['Age', 'Gender', 'Cholesterol', 'Blood Pressure', 'Heart Rate',
       'Smoking', 'Alcohol Intake', 'Exercise Hours', 'Family History',
       'Diabetes', 'Obesity', 'Stress Level', 'Blood Sugar',
       'Exercise Induced Angina', 'Chest Pain Type', 'Heart Disease'],
      dtype='object')

these are all the columns of dataset

In [68]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Age                      1000 non-null   int64 
 1   Gender                   1000 non-null   object
 2   Cholesterol              1000 non-null   int64 
 3   Blood Pressure           1000 non-null   int64 
 4   Heart Rate               1000 non-null   int64 
 5   Smoking                  1000 non-null   object
 6   Alcohol Intake           660 non-null    object
 7   Exercise Hours           1000 non-null   int64 
 8   Family History           1000 non-null   object
 9   Diabetes                 1000 non-null   object
 10  Obesity                  1000 non-null   object
 11  Stress Level             1000 non-null   int64 
 12  Blood Sugar              1000 non-null   int64 
 13  Exercise Induced Angina  1000 non-null   object
 14  Chest Pain Type          1000 non-null   

data type of each column is mentioned above.

3. How many numerical and categorical columns are there?

In [69]:
data.dtypes.value_counts()

int64     8
object    8
Name: count, dtype: int64

there are:
    numerical columns = 8,
    categorical cloumns = 8

4. What are the basic statistics (mean, median, min, max) for all numerical columns?

In [70]:
data.describe()

,Age,Cholesterol,Blood Pressure,Heart Rate,Exercise Hours,Stress Level,Blood Sugar,Heart Disease
count,1000.000000,1000.000000,1000.0000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,52.293000,249.939000,135.2810,79.204000,4.529000,5.646000,134.941000,0.392000
std,15.727126,57.914673,26.3883,11.486092,2.934241,2.831024,36.699624,0.488441
min,25.000000,150.000000,90.0000,60.000000,0.000000,1.000000,70.000000,0.000000
25%,39.000000,200.000000,112.7500,70.000000,2.000000,3.000000,104.000000,0.000000
50%,52.000000,248.000000,136.0000,79.000000,4.500000,6.000000,135.000000,0.000000
75%,66.000000,299.000000,159.0000,89.000000,7.000000,8.000000,167.000000,1.000000
max,79.000000,349.000000,179.0000,99.000000,9.000000,10.000000,199.000000,1.000000


we can see mean, median,min and max for all numerical column

5. Are there any missing values? If yes, in which columns and how many?

In [71]:
data.isnull().sum()

Age                          0
Gender                       0
Cholesterol                  0
Blood Pressure               0
Heart Rate                   0
Smoking                      0
Alcohol Intake             340
Exercise Hours               0
Family History               0
Diabetes                     0
Obesity                      0
Stress Level                 0
Blood Sugar                  0
Exercise Induced Angina      0
Chest Pain Type              0
Heart Disease                0
dtype: int64

yes, there are 340 missing value in Alcohol Intake column.

6. Are there any duplicate rows?

In [72]:
data.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
995    False
996    False
997    False
998    False
999    False
Length: 1000, dtype: bool

No, there are no duplicated data.

7. What is the class balance — how many patients have heart disease vs not?

In [73]:
data.sample(5)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
541,70,Female,331,178,93,Never,Moderate,4,No,Yes,Yes,6,111,Yes,Non-anginal Pain,1
200,57,Male,267,105,75,Current,NaN,6,No,No,No,3,197,No,Asymptomatic,1
233,27,Male,191,95,95,Never,NaN,5,Yes,Yes,No,7,187,Yes,Typical Angina,0
188,53,Male,265,93,82,Current,Heavy,2,No,No,No,7,113,Yes,Asymptomatic,1
318,57,Male,317,169,76,Current,NaN,1,Yes,No,Yes,7,157,Yes,Non-anginal Pain,1


In [74]:
data['Heart Disease'].value_counts()

Heart Disease
0    608
1    392
Name: count, dtype: int64

Heart disease patient = 392

Non Heard disease patient = 608

8. What is the age range of patients in the dataset?

In [75]:
youngest_age = data['Age'].min()
oldest_age = data['Age'].max()
age_range = oldest_age - youngest_age
print(f"Youngest Age = {youngest_age}")
print(f"Oldest Age = {oldest_age}")
print(f"Range of Age = {oldest_age - youngest_age}")

Youngest Age = 25
Oldest Age = 79
Range of Age = 54


🟠 Section 2 — Data Cleaning (Q9–Q16)

9. Handle the missing values in the Alcohol Intake column — what strategy will you use and why?

In [76]:
data.sample(5)

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
392,78,Female,199,130,98,Never,NaN,8,No,Yes,No,3,166,No,Atypical Angina,0
759,71,Female,161,175,78,Never,Heavy,1,No,Yes,Yes,2,114,No,Non-anginal Pain,0
864,71,Female,269,120,63,Never,Heavy,7,Yes,No,Yes,3,116,No,Asymptomatic,1
510,79,Male,279,98,97,Former,Moderate,9,No,No,No,1,148,Yes,Typical Angina,1
791,49,Female,259,143,94,Never,Moderate,5,Yes,No,Yes,1,172,No,Typical Angina,0


In [77]:
data.isna().sum()

Age                          0
Gender                       0
Cholesterol                  0
Blood Pressure               0
Heart Rate                   0
Smoking                      0
Alcohol Intake             340
Exercise Hours               0
Family History               0
Diabetes                     0
Obesity                      0
Stress Level                 0
Blood Sugar                  0
Exercise Induced Angina      0
Chest Pain Type              0
Heart Disease                0
dtype: int64

In [78]:
data['Alcohol Intake'].value_counts()

Alcohol Intake
Heavy       346
Moderate    314
Name: count, dtype: int64

Alcohol Intake column have 340 missing valuse which is about one third of our dataset. so dropping is recommended.

It is categorical column so so we can not find the mean or median to fill missing value. 

Most of the people do not drink so we will fill the missing value with 'Never'

In [79]:
data['Alcohol Intake'].fillna('Never', inplace=True)

/var/folders/d8/83r4k_r15tv7zl64z94ffs640000gn/T/ipykernel_21966/1296818874.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  data['Alcohol Intake'].fillna('Never', inplace=True)


In [80]:
data.isna().sum()

Age                        0
Gender                     0
Cholesterol                0
Blood Pressure             0
Heart Rate                 0
Smoking                    0
Alcohol Intake             0
Exercise Hours             0
Family History             0
Diabetes                   0
Obesity                    0
Stress Level               0
Blood Sugar                0
Exercise Induced Angina    0
Chest Pain Type            0
Heart Disease              0
dtype: int64

no any missing values now

In [81]:
data['Alcohol Intake'].value_counts()

Alcohol Intake
Heavy       346
Never       340
Moderate    314
Name: count, dtype: int64

Now Alcohol Intake column changed to 

Moderate drinker = 314

Never = 340 

Heavy drinker = 346

10. Check for any impossible or suspicious values in numerical columns (e.g. Cholesterol = 0, Heart Rate = 0).

In [89]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Age                      1000 non-null   int64 
 1   Gender                   1000 non-null   object
 2   Cholesterol              1000 non-null   int64 
 3   Blood Pressure           1000 non-null   int64 
 4   Heart Rate               1000 non-null   int64 
 5   Smoking                  1000 non-null   object
 6   Alcohol Intake           1000 non-null   object
 7   Exercise Hours           1000 non-null   int64 
 8   Family History           1000 non-null   object
 9   Diabetes                 1000 non-null   object
 10  Obesity                  1000 non-null   object
 11  Stress Level             1000 non-null   int64 
 12  Blood Sugar              1000 non-null   int64 
 13  Exercise Induced Angina  1000 non-null   object
 14  Chest Pain Type          1000 non-null   

In [93]:
print("Impossible values in age = ",(data['Age']==0).sum())
print("Impossible value in Cholesterol = ",(data['Cholesterol']==0).sum())
print("Impossible values in Blood Pressure  =",(data['Blood Pressure']==0).sum())
print("impossible values in Heart Rate = ",(data['Heart Rate']==0).sum())
print("impossible values in Blood sugar = ",(data['Blood Sugar']==0).sum())


Impossible values in age =  0
Impossible value in Cholesterol =  0
Impossible values in Blood Pressure  = 0
impossible values in Heart Rate =  0
impossible values in Blood sugar =  0


No any impossible or suspicious values in numerical columns like  Age, Cholesterol, Blood Pressure, Heart Rate, Blood sugar

11. Convert Heart Disease column (0/1) into a readable label column (No/Yes) for visualization purposes.

In [94]:
data['Heart Disease'].map({0: 'No', 1: 'Yes'})

0      Yes
1       No
2      Yes
3       No
4       No
      ... 
995    Yes
996    Yes
997     No
998    Yes
999    Yes
Name: Heart Disease, Length: 1000, dtype: object

In [95]:
data['Heart Disease']=data['Heart Disease'].map({0: 'No', 1: 'Yes'})

In [97]:
data.sample()

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease
413,57,Female,229,179,63,Never,Heavy,2,No,Yes,No,7,144,No,Atypical Angina,Yes


12. Are the categorical columns (Gender, Smoking, Chest Pain Type, etc.) consistently formatted with no spelling errors?

In [99]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   Age                      1000 non-null   int64 
 1   Gender                   1000 non-null   object
 2   Cholesterol              1000 non-null   int64 
 3   Blood Pressure           1000 non-null   int64 
 4   Heart Rate               1000 non-null   int64 
 5   Smoking                  1000 non-null   object
 6   Alcohol Intake           1000 non-null   object
 7   Exercise Hours           1000 non-null   int64 
 8   Family History           1000 non-null   object
 9   Diabetes                 1000 non-null   object
 10  Obesity                  1000 non-null   object
 11  Stress Level             1000 non-null   int64 
 12  Blood Sugar              1000 non-null   int64 
 13  Exercise Induced Angina  1000 non-null   object
 14  Chest Pain Type          1000 non-null   

In [ ]:
print(data['Gender'].unique())
print(data['Smoking'].unique())
print(data['Alcohol Intake'].unique())
print(data['Family History'].unique())
print(data['Diabetes'].unique())
print(data['Obesity'].unique())
print(data['Exercise Induced Angina'].unique())
print(data['Chest Pain Type'].unique())
print(data['Heart Disease'].unique())

['Female' 'Male']
['Current' 'Never' 'Former']
['Heavy' 'Never' 'Moderate']
['No' 'Yes']
['No' 'Yes']
['Yes' 'No']
['Yes' 'No']
['Atypical Angina' 'Typical Angina' 'Non-anginal Pain' 'Asymptomatic']
['Yes' 'No']


No, the categorical columns (Gender, Smoking, Chest Pain Type, etc.) consistently formatted with no spelling errors.

13. Detect outliers in Cholesterol, Blood Sugar, and Heart Rate using the IQR method — how many outliers exist?

In [105]:
# function to check and handle outliers

def handle_outliers_with_median(df, column_name):
    # 1. Calculate IQR statistics using clean, local variable names
    Q1 = df[column_name].quantile(0.25)
    Q3 = df[column_name].quantile(0.75)
    IQR = Q3 - Q1
    
    # 2. Define bounds
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR
    
    # 3. Find and count outliers
    outliers_mask = (df[column_name] < lower_bound) | (df[column_name] > upper_bound)
    outlier_count = outliers_mask.sum()
    print(f"Found {outlier_count} outliers in '{column_name}'")
    
    # 4. Replace outliers with the column median
    median_val = df[column_name].median()
    df.loc[outliers_mask, column_name] = median_val
    
    return df


In [109]:
data = handle_outliers_with_median(data,'Cholesterol')
data = handle_outliers_with_median(data,'Blood Sugar')
data = handle_outliers_with_median(data,'Heart Rate')

Found 0 outliers in 'Cholesterol'
Found 0 outliers in 'Blood Sugar'
Found 0 outliers in 'Heart Rate'


14. Create a new column Age_Group — Young (25–40), Middle-Aged (41–60), Senior (61–79).

In [113]:
data.loc[((data['Age']>=25) & (data['Age']<=40)),'Age_Group'] = 'Young'
data.loc[((data['Age']>=41) & (data['Age']<=60)),'Age_Group'] = 'Middle-Aged'
data.loc[(data['Age']>=61) & (data['Age']<=79),'Age_Group'] = 'Senior'

In [117]:
data['Age_Group'].value_counts()

Age_Group
Middle-Aged    372
Senior         353
Young          275
Name: count, dtype: int64

15. Create a new column Cholesterol_Category — Normal (<200), Borderline (200–239), High (≥240).

In [118]:
# data.loc[condition,'column_name'] = "catagory_name"
data.loc[(data['Cholesterol']<200),'Cholesterol_Category'] = 'Normal'
data.loc[(data['Cholesterol']>=200) & (data['Cholesterol']<=239), 'Cholesterol_Category'] = 'Borderline'
data.loc[(data['Cholesterol']>=240),'Cholesterol_Category'] = 'High'

In [120]:
data['Cholesterol_Category'].value_counts()

Cholesterol_Category
High          550
Normal        246
Borderline    204
Name: count, dtype: int64

16. Create a new column BP_Category — Normal (<120), Elevated (120–129), High (≥130).

In [121]:
data.loc[(data['Blood Pressure']<120),'BP_Category'] = 'Normal'
data.loc[(data['Blood Pressure']>=120) & (data['Blood Pressure']<=129),'BP_Category'] = 'Elevated'
data.loc[(data['Blood Pressure']>=130), 'BP_Category'] = 'High'

In [122]:
data['BP_Category'].value_counts()

BP_Category
High        571
Normal      333
Elevated     96
Name: count, dtype: int64

🟡 Section 3 — Data Manipulation (Q17–Q24)

17. What is the average age of patients with and without heart disease?

In [123]:
data['Heart Disease'].value_counts()

Heart Disease
No     608
Yes    392
Name: count, dtype: int64

In [134]:
data.groupby('Heart Disease')['Age'].mean().round(2)

Heart Disease
No     44.13
Yes    64.96
Name: Age, dtype: float64

the average age of patient with heart dieases = 64.96

the average age of patient without heard disease = 44.13

18. What percentage of male vs female patients have heart disease?

In [140]:
(data.groupby("Heart Disease")['Gender'].value_counts(normalize=True)*100).round(2)

Heart Disease  Gender
No             Female    51.48
               Male      48.52
Yes            Male      51.53
               Female    48.47
Name: proportion, dtype: float64

51.53% male have heart disease

48.47% female have heart disease


19. Which smoking status group (Current/Former/Never) has the highest rate of heart disease?

In [141]:
data.sample()

,Age,Gender,Cholesterol,Blood Pressure,Heart Rate,Smoking,Alcohol Intake,Exercise Hours,Family History,Diabetes,Obesity,Stress Level,Blood Sugar,Exercise Induced Angina,Chest Pain Type,Heart Disease,Age_Group,Cholesterol_Category,BP_Category
264,42,Female,278,158,77,Never,Heavy,4,No,Yes,Yes,4,140,Yes,Typical Angina,No,Middle-Aged,High,High


In [145]:
(data.groupby('Heart Disease')['Smoking'].value_counts(normalize=True) * 100 ).round(2)

Heart Disease  Smoking
No             Former     36.02
               Never      32.07
               Current    31.91
Yes            Never      36.48
               Current    36.22
               Former     27.30
Name: proportion, dtype: float64

Never smoking status group has the highest rate of heart disease.

currently smoking status group has the 2nd highest rate of heart disease.

20. Do patients with diabetes have a higher rate of heart disease compared to non-diabetic patients?

In [149]:
(data.groupby('Heart Disease')['Diabetes'].value_counts(normalize=True) * 100).round(2)

Heart Disease  Diabetes
No             No          50.16
               Yes         49.84
Yes            Yes         51.53
               No          48.47
Name: proportion, dtype: float64

yes, patient with diabetes have a higher rate of heart disease compared to non-diabetic patients

21. What is the average Cholesterol, Blood Pressure, and Stress Level for patients with heart disease vs without?

In [ ]:
data.groupby('Heart Disease')['Cholesterol'].mean().round(2)

Average cholesterol with and without heart disease:


Heart Disease
No     232.97
Yes    276.26
Name: Cholesterol, dtype: float64

average cholesterol with heart disease = 276.26

average cholesterol without heart disease = 232.97

In [153]:
data.groupby('Heart Disease')['Blood Pressure'].mean().round(2)

Heart Disease
No     135.13
Yes    135.51
Name: Blood Pressure, dtype: float64

Average Blood Pressure with heart disease = 135.51

Average Blood Pressure without heart disease = 135.13



In [154]:
data.groupby('Heart Disease')['Stress Level'].mean().round(2)

Heart Disease
No     5.63
Yes    5.67
Name: Stress Level, dtype: float64

Average Stress level with heart disease = 5.67

Average Stress level without heart disease = 5.63

22. Which Chest Pain Type is most commonly associated with heart disease?

In [156]:
(data.groupby('Heart Disease')['Chest Pain Type'].value_counts(normalize=True) * 100).round(2)

Heart Disease  Chest Pain Type 
No             Non-anginal Pain    27.30
               Typical Angina      24.67
               Asymptomatic        24.01
               Atypical Angina     24.01
Yes            Asymptomatic        26.02
               Atypical Angina     25.51
               Typical Angina      25.51
               Non-anginal Pain    22.96
Name: proportion, dtype: float64

Asymptomatic type of chest pain is most commanly associated with heart disease.

23. Do patients with a Family History of heart disease have a significantly higher rate of the condition?

In [159]:
data.groupby('Heart Disease')['Family History'].value_counts()

Heart Disease  Family History
No             No                312
               Yes               296
Yes            Yes               203
               No                189
Name: count, dtype: int64